make.paws/tests/testthat/test_handlers_restxml.R

```diff
@@ -0,0 +1,664 @@
+    )
+    return(populate(args, interface))
+  }
+
+  input <- op_input_test(
+    ListParam = list(
+      "one",
+      "two",
+      "three"
+    )
+  )
+  req <- new_request(svc, op_test, input, NULL)
+  req <- build(req)
+  r <- req$body
+  expect_equal(r, '<OperationRequest xmlns="https://foo/"><AlternateName xmlns="https://foo/"><NotMember xmlns="https://foo/">one</NotMember><NotMember xmlns="https://foo/">two</NotMember><NotMember xmlns="https://foo/">three</NotMember></AlternateName></OperationRequest>')
+})
+
+test_that("Flattened List Case1", {
+  op_test <- Operation(name = "OperationRequest")
+  op_input_test <- function(ListParam) {
+    args <- list(ListParam = ListParam)
+    interface <- Structure(
+      ListParam = List(
+        Scalar(type = "string"),
+        .attrs = list(flattened = "true")
+      ),
+      .attrs = list(locationName = "OperationRequest", xmlURI = "https://foo/")
+    )
+    return(populate(args, interface))
+  }
+
+  input <- op_input_test(
+    ListParam = list(
+      "one",
+      "two",
+      "three"
+    )
+  )
+  req <- new_request(svc, op_test, input, NULL)
+  req <- build(req)
+  r <- req$body
+  expect_equal(r, '<OperationRequest xmlns="https://foo/"><ListParam xmlns="https://foo/">one</ListParam><ListParam xmlns="https://foo/">two</ListParam><ListParam xmlns="https://foo/">three</ListParam></OperationRequest>')
+})
+
+test_that("Flattened List with LocationName Case1", {
+  op_test <- Operation(name = "OperationRequest")
+  op_input_test <- function(ListParam) {
+    args <- list(ListParam = ListParam)
+    interface <- Structure(
+      ListParam = List(
+        Scalar(type = "string"),
+        .attrs = list(flattened = "true", locationName = "item")
+      ),
+      .attrs = list(locationName = "OperationRequest", xmlURI = "https://foo/")
+    )
+    return(populate(args, interface))
+  }
+
+  input <- op_input_test(
+    ListParam = list(
+      "one",
+      "two",
+      "three"
+    )
+  )
+  req <- new_request(svc, op_test, input, NULL)
+  req <- build(req)
+  r <- req$body
+  expect_equal(r, '<OperationRequest xmlns="https://foo/"><item xmlns="https://foo/">one</item><item xmlns="https://foo/">two</item><item xmlns="https://foo/">three</item></OperationRequest>')
+})
+
+test_that("List of Structures Case1", {
+  op_test <- Operation(name = "OperationName")
+  op_input_test <- function(ListParam) {
+    args <- list(ListParam = ListParam)
+    interface <- Structure(
+        ListParam = List(
+          Structure(
+            Scalar(type = "string", .attrs = list(locationName = "value"))
+          ),
+          .attrs = list(flattened = "true", locationName = "item")
+        ),
+        .attrs = list(locationName = "OperationRequest",
+                      xmlURI = "https://foo/")
+    )
+    return(populate(args, interface))
+  }
+
+  input <- op_input_test(
+      ListParam = list(
+        list(
+          value = list("one")
```

Comment:
and these should probably be only `"one"`, etc.

Reply:
It expected an input in the form of
```
input <- op_input_test(
      ListParam = list(
          list(Element = "one"),
          ...
```

Reply:
Ah right. Yes.